examples/launch_configuration/main.tf

```diff
@@ -0,0 +1,339 @@
+        },
+        "Effect": "Allow",
+        "Sid": ""
+      }
+    ]
+  }
+  EOT
+}
+
+module "alb_http_sg" {
+  source  = "terraform-aws-modules/security-group/aws//modules/http-80"
+  version = "~> 3"
+
+  name        = "${local.name}-alb-http"
+  vpc_id      = module.vpc.vpc_id
+  description = "Security group for ${local.name}"
+
+  ingress_cidr_blocks = ["0.0.0.0/0"]
+
+  tags = local.tags_as_map
+}
+
+module "alb" {
+  source  = "terraform-aws-modules/alb/aws"
+  version = "~> 5"
+
+  name = local.name
+
+  vpc_id          = module.vpc.vpc_id
+  subnets         = module.vpc.public_subnets
+  security_groups = [module.alb_http_sg.this_security_group_id]
+
+  http_tcp_listeners = [
+    {
+      port               = 80
+      protocol           = "HTTP"
+      target_group_index = 0
+    }
+  ]
+
+  target_groups = [
+    {
+      name             = local.name
+      backend_protocol = "HTTP"
+      backend_port     = 80
+      target_type      = "instance"
+    },
+  ]
+
+  tags = local.tags_as_map
+}
+
+################################################################################
+# Disabled
+################################################################################
+
+module "disabled" {
+  source = "../../"
+
+  create_asg = false
+
+  # Autoscaling group
+  name = "disabled-${local.name}"
+}
+
+################################################################################
+# Default
+################################################################################
+
+module "default" {
+  source = "../../"
+
+  # Autoscaling group
+  name = "default-${local.name}"
+
+  vpc_zone_identifier = module.vpc.private_subnets
+  min_size            = 0
+  max_size            = 1
+  desired_capacity    = 1
+
+  # Launch configuration
+  use_lc    = true
+  create_lc = true
+
+  image_id      = data.aws_ami.amazon_linux.id
+  instance_type = "t3.micro"
+
+  tags        = local.tags
+  tags_as_map = local.tags_as_map
+}
+
+################################################################################
+# External
+################################################################################
+
+resource "aws_launch_configuration" "this" {
+  name_prefix   = "external-${local.name}-"
+  image_id      = data.aws_ami.amazon_linux.id
+  instance_type = "t3.micro"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+module "external" {
+  source = "../../"
+
+  # Autoscaling group
+  name = "external-${local.name}"
+
+  vpc_zone_identifier = module.vpc.private_subnets
+  min_size            = 0
+  max_size            = 1
+  desired_capacity    = 1
+
+  # Launch config
+  use_lc               = true
+  launch_configuration = aws_launch_configuration.this.name
+
+  tags        = local.tags
+  tags_as_map = local.tags_as_map
+}
+
+################################################################################
+# Complete
+################################################################################
+
+module "complete" {
```

Comment:
It is great to have detailed examples and I propose to combine all of them into 1 set of examples because there is a lot of overall between LC and LT (eg, extra resources like VPC, SG, ALB, IAM, disabled, default, outputs, locals).

I think having 3 complete modules for LC/LT/mixed, one for default, one disabled is enough.

Having less code to maintain is better. You know :)

Reply:
sounds good - just to be clear, would that all be under one example directory/project?

Reply:
I think, the one combined is enough.

Reply:
fixed in https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/139/commits/50eb54f1fe485641bf3517661cbbc0dc7c37acd8